src/VisualStudio/IntegrationTest/TestUtilities/InProcess/SolutionExplorer_InProc.cs

```diff
@@ -59,6 +59,23 @@ public static SolutionExplorer_InProc Create()
             };
         }
 
+        public void AddMetadataReference(string assemblyName, string projectName)
+        {
+            var referenceAssemblyFullPath = System.Reflection.Assembly.Load(assemblyName).Location;
+            var project = GetProject(projectName);
+            ((VSProject)project.Object).References.Add(referenceAssemblyFullPath);
+        }
+
+        public void RemoveMetadataReference(string assemblyName, string projectName)
+        {
+            var project = GetProject(projectName);
+            if (project.Object is VSProject vsproject)
```

Comment:
Please throw if the project isn't valid, since that means you're passing the wrong name.